tracer/src/Datadog.Trace/Agent/MessagePack/SpanMessagePackFormatter.cs

```diff
@@ -639,8 +658,20 @@ internal TraceTagWriter(SpanMessagePackFormatter formatter, ITagProcessor[] tagP
             [MethodImpl(MethodImplOptions.AggressiveInlining)]
             public void Next(KeyValuePair<string, string> item)
             {
-                _formatter.WriteTag(ref Bytes, ref Offset, item.Key, item.Value, _tagProcessors);
-                Count++;
+                if (_isFirstSpanInChunk &&
+                    item.Key.StartsWith(TagPropagation.PropagatedTagPrefix, StringComparison.Ordinal))
```

Comment:
_Probably_ unnecessary, but [I believe `Span<T>.StartWith()` outperforms `String.StartsWith()](https://tjaddison.com/blog/2021/06/run-benchmarks-using-multiple-runtimes-with-benchmarkdotnet/)` so you _may_ want to consider specialising for .NET Coreapp, e.g.

```suggestion
#if NETCOREAPP
                if (_isFirstSpanInChunk &&
                    item.Key.AsSpan().StartsWith(TagPropagation.PropagatedTagPrefix.AsSpan()))
#else
                if (_isFirstSpanInChunk &&
                    item.Key.StartsWith(TagPropagation.PropagatedTagPrefix, StringComparison.Ordinal))
#endif
```

Also note that as StringComparison.Ordinal